vocode/streaming/utils/worker.py

```diff
@@ -66,7 +66,7 @@ def consume_nonblocking(self, item: WorkerInputType):
     async def _run_loop(self):
         raise NotImplementedError
 
-    def terminate(self):
+    async def terminate(self):
         if self.worker_task:
             return self.worker_task.cancel()
```

Comment:
Huh. So worker task was always async but we didn't wait on it before

Reply:
we're still not waiting on it here - `cancel` doesn't need to be awaited, it only needs to be called from an "async context" (which means within an async program)